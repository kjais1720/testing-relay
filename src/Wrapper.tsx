import { LocalizationProvider } from '@material-ui/pickers'
import MomentUtils from '@material-ui/pickers/adapter/moment'
import { HashRouter } from '@saastack/router'
import React from 'react'
import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles'
import { getThemeOptions } from '@saastack/utils/localization/theme'
import GlobalStyles from './GlobalStyles'
import { i18n } from '@lingui/core'
import { I18nProvider } from '@lingui/react'

i18n.load({
    'en-US': {
        messages: { Save: 'Save' },
    },
})

const Wrapper: React.FC = ({ children }) => (
    <I18nProvider i18n={i18n} language="en-US">
        <LocalizationProvider dateAdapter={MomentUtils}>
            <HashRouter>
                <ThemeProvider theme={createMuiTheme(getThemeOptions())}>
                    <GlobalStyles />
                    {children}
                </ThemeProvider>
            </HashRouter>
        </LocalizationProvider>
    </I18nProvider>
)

export default Wrapper
