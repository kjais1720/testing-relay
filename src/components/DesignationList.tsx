import { ListContainer } from '@saastack/layouts/containers'
import { ListContainerProps } from '@saastack/layouts/types'
import { Trans } from '@lingui/macro'
import { ListItemText } from '@material-ui/core'
import React from 'react'
import { createFragmentContainer, graphql } from 'react-relay'
import { DesignationList_designations } from '../__generated__/DesignationList_designations.graphql'
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction'
import { ActionItem, Actions } from '@saastack/components'
import { DeleteOutlined, EditOutlined } from '@material-ui/icons'
import useNavigate from '@saastack/router/useNavigate'

interface Props extends Omit<ListContainerProps<DesignationList_designations>, 'items'> {
    designations: DesignationList_designations
}

const DesignationList: React.FC<Props> = ({ designations, ...props }) => {
    const navigate = useNavigate()
    const listProps = {
        render: (e: DesignationList_designations[0]) => {
            const actions: ActionItem[] = [
                {
                    title: <Trans>Update</Trans>,
                    icon: EditOutlined,
                    onClick: () => navigate(`${window.btoa(e.id)}/update`),
                },
                {
                    title: <Trans>Delete</Trans>,
                    icon: DeleteOutlined,
                    onClick: () => navigate(`${window.btoa(e.id)}/delete`),
                },
            ]
            return [
                <ListItemText key="name" primary={e.name} secondary={e.description}>
                </ListItemText>,
                <ListItemSecondaryAction key="action">
                    <Actions items={actions}/>
                </ListItemSecondaryAction>,

            ]
        },
    }
    return <ListContainer<DesignationList_designations> items={designations} listProps={listProps}  {...props}/>
}

export default createFragmentContainer(DesignationList, {
    designations: graphql`
        fragment DesignationList_designations on Designation @relay(plural: true) {
            id,
            name,
            description
        }
    `,
})
