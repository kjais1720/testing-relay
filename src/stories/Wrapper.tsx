import { Mutable, TemplateLoader } from '@saastack/core'
import { withEnvironment, withQuery, WithQueryProps } from '@saastack/relay'
import { ModernTemplate } from '@saastack/templates/templates'
import { compose } from '@saastack/utils'
import React from 'react'
import { graphql } from 'relay-runtime'
import { WrapperQuery } from '../__generated__/WrapperQuery.graphql'

const query = graphql`
    query WrapperQuery {
        viewer {
            id
            firstName
            lastName
            email
            profileImage {
                thumbImage
            }
            preferences {
                dateFormat
                language
                timeFormat
                timezone
                uIInfo
            }
            userRoles {
                role {
                    levelDetails {
                        ... on Location {
                            id
                        }
                        ... on Company {
                            id
                        }
                        ... on Group {
                            id
                        }
                    }
                    role {
                        moduleRoles {
                            name
                        }
                    }
                }
            }
            groups {
                id
                name
                companies {
                    id
                    title
                    displayName
                    metadata
                    slug
                    gallery {
                        default {
                            thumbImage
                        }
                    }
                    address {
                        country
                    }
                    companySettings {
                        navMenus
                        aliases(locale: "en-US")
                    }
                    apps {
                        id
                        appTypeId
                        name
                        active
                        serviceModules
                    }
                    locations(first: 500) {
                        edges {
                            node {
                                id
                                name
                                preference {
                                    currency
                                }
                                slug
                            }
                        }
                    }
                }
            }
        }
    }
`

interface Props extends WithQueryProps<WrapperQuery> {
}

const Wrapper: React.FC<Props> = props => {
    return (
        <TemplateLoader template={ModernTemplate} getLocale={() => Promise.resolve({ messages: {} })} viewer={props.query?.viewer as Mutable<WrapperQuery['response']['viewer']>}>
            {props.children}
        </TemplateLoader>
    )
}


export default compose(
    withEnvironment({ graphqlUrl: process.env.REACT_APP_GRAPHQL_URL }),
    withQuery<WrapperQuery>({ query, Loading: () => <>Loading logged in user...</> }),
)(Wrapper)
