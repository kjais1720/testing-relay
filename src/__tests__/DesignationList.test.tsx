import React from 'react'
import { QueryRenderer, graphql } from 'react-relay'
import { render, screen } from '@testing-library/react'
import DesignationList from '../components/DesignationList'
import Wrapper from '../Wrapper'
import { createMockEnvironment, MockPayloadGenerator, MockEnvironment } from 'relay-test-utils'
import reactTestRenderer from 'react-test-renderer'
import { DesignationMaster_designations} from "../__generated__/DesignationMaster_designations.graphql"
import "@testing-library/jest-dom"
const TestRenderer = ({environment}:{environment:MockEnvironment})=>(
  <QueryRenderer
      environment={environment}
      query={graphql`
      query DesignationPageQuery @relay_test_operation{
          ...DesignationMaster_designations @arguments(parent: "$parent")
      }
  `}
      variables={{ parent: 'tredrdhhh' }}
      render={({ error, props }) => {
          if (props) {
              console.log(props)
              return <div>jest</div>
          } else if (error) {
              return error.message
          }
          return <div data-testId="loader">
            Loading...
          </div>
      }}
  />
)

fit("Should show loader while data is being fetched",()=>{
  const environment = createMockEnvironment()
  const renderer = reactTestRenderer.create(<TestRenderer environment={environment}/>)
  expect(
    renderer.root.find(node=>node.props["data-testId"] === "loader")
  ).toBeDefined()
})

it('Fragment Container', () => {
    const environment = createMockEnvironment()
    const renderer = reactTestRenderer.create(<TestRenderer environment={environment}/>)

    environment.mock.resolveMostRecentOperation((operation) =>
        MockPayloadGenerator.generate(operation, {
            ID(_, generateId) {
                return `my-id-${generateId()}`
            },
            String(context) {
                return 'Satya'
            },
        })
    )

    expect(renderer).toMatchSnapshot()
})
