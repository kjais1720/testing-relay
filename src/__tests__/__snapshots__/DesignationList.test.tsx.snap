// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fragment Container 1`] = `
<div>
  jest
</div>
`;
